test/test_from_exception.cpp

```diff
@@ -31,7 +31,15 @@ BOOST_NOINLINE BOOST_SYMBOL_VISIBLE void in_test_throw_1(const char* msg) {
 
 BOOST_NOINLINE BOOST_SYMBOL_VISIBLE void in_test_throw_2(const char* msg) {
   std::string new_msg{msg};
-  throw std::runtime_error(new_msg);
+  throw std::logic_error(new_msg);
```

Comment:
Without this change, `in_test_throw_2` and `in_test_throw_1` have identical function body, then seems MSVC merges this two functions, causing stacktrace shows wrong function name.